shdlc/sensirion_shdlc.c

```diff
@@ -201,3 +201,174 @@ int16_t sensirion_shdlc_rx(uint8_t max_data_len,
 
     return 0;
 }
+
+static void sensirion_shdlc_stuff_byte(struct sensirion_shdlc_buffer* tx_frame,
+                                       uint8_t data) {
+    switch (data) {
+        case 0x11:
+        case 0x13:
+        case 0x7d:
+        case 0x7e:
+            // byte stuffing is done by inserting 0x7d and inverting bit 5
+            tx_frame->data[tx_frame->offset++] = 0x7d;
+            tx_frame->data[tx_frame->offset++] = data ^ (1 << 5);
+            return;
+        default:
+            tx_frame->data[tx_frame->offset++] = data;
+            return;
+    }
+}
+
+void sensirion_shdlc_add_uint8_t_to_frame(
+    struct sensirion_shdlc_buffer* tx_frame, uint8_t data) {
+    sensirion_shdlc_stuff_byte(tx_frame, data);
```

Comment:
hard to read, might benefit from a blank line above (also for other instances of functions with indented arguments in this PR)